### PR TITLE
Fixes #720 - check_snmp numeric OID name bug

### LINKED ISSUE #720: check_snmp plugin is not parsing the OID data correctly for integers.
The check_snmp plugin uses the snmpget application to poll the data and snmpget returns the OID name and the value when it runs.

The plugin is parsing that data and if the OID name has numeric data in it, the plugin uses that incorrectly instead of the value of the OID data.

If you see below, the OID name has 1000 in it but the value of the OID is 2 and the thresholds are using 1000 from the name instead of 2 for the data.

/usr/local/nagios/libexec/check_snmp -H 192.168.112.130 -C public -P 2c -o ipv6IpForwarding.0  -v -v -v -v -w 2 -c 6
/usr/bin/snmpget -Le -t 3 -r 5 -m ALL -v 2c [context] [authpriv] 192.168.112.130:161 ipv6IpForwarding.0
IP-MIB::ipv6IpForwarding.0 = INTEGER: notForward1000ing(2)
Processing oid 1 (line 1)
  oidname: IP-MIB::ipv6IpForwarding.0
  response: INTEGER: notForward1000ing(2)

SNMP CRITICAL - *1000* | IP-MIB::ipv6IpForwarding.0=1000;2;6

### PR DESCRIPTION
**Pull request by nagiosgwesterman**
Testing Plan:
- Tested plugin using this command: `/usr/local/nagios/libexec/check_snmp -H 192.168.223.19 -C public -P 2c -o ipv6IpForwarding.0  -v -v -v -v -w 2 -c 6`
- Before making the changes, this command returns:
```
/usr/bin/snmpget -Le -t 3 -r 5 -m ALL -v 2c [context] [authpriv] 192.168.223.19:161 ipv6IpForwarding.0
IP-MIB::ipv6IpForwarding.0 = INTEGER: notForward1000ing(2)
Processing oid 1 (line 1)
oidname: IP-MIB::ipv6IpForwarding.0
response: INTEGER: notForward1000ing(2)

SNMP CRITICAL - 1000 | IP-MIB::ipv6IpForwarding.0=1000;2;6
```
- Notice that the response uses the value 1000 which is in the OID name rather than the actual OID value (which is 2).
- After making the changes, this command returns:
```
/usr/bin/snmpget -Le -t 3 -r 5 -m ALL -v 2c [context] [authpriv] 192.168.223.19:161 ipv6IpForwarding.0
IP-MIB::ipv6IpForwarding.0 = INTEGER: notForward1000ing(2)
Processing oid 1 (line 1)
oidname: IP-MIB::ipv6IpForwarding.0
response: INTEGER: notForward1000ing(2)

SNMP OK - 2 | IP-MIB::ipv6IpForwarding.0=2;2;6
```
- And it correctly gets and uses the value 2.

### CODE REVIEW

**Comment by sawolf** [comment]
Hey Griffin, the code looks fine to me, but there a couple of things here:
 - First, Github doesn't link to issues in the title, so: #720 
 - Second, please write out the testing plan you followed prior to making this pull request

File: plugins/check_snmp.c [hunk 1]
```diff
@@ -496,8 +496,10 @@ main (int argc, char **argv)
...
 		/* Make some special values,like Timeticks numeric only if a threshold is defined */
 		if (thlds[i]->warning || thlds[i]->critical || calculate_rate || is_ticks || offset != 0.0 || multiplier != 1.0) {
-			ptr = strpbrk (show, "-0123456789");
-
+			/* Find the first instance of the '(' character - the value of the OID should be contained in parens */
+			ptr = strpbrk(show, "(");
+			ptr++;
+			
 			if (ptr == NULL)
 				die (STATE_UNKNOWN,_("No valid data returned (%s)\n"), show);
```
Comment (sawolf) [inline plugins/check_snmp.c:504]: Will `ptr` ever by `NULL` by the time we're checking it?

Comment (nagiosgwesterman) [reply plugins/check_snmp.c:504]: You're right, that check should be above the ptr increment.

Comment (nagiosgwesterman) [reply plugins/check_snmp.c]: Fixed

**Comment by sawolf** [comment]
Alright, this looks good to me now. Thank you!

**Pull request merged**